catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -155,18 +212,22 @@ private void fetchNextResults() {
             queryCopy.setStartIndex(currentIndex);
 
             try {
-                SourceResponse response = queryFunction.query(queryRequestCopy);
+                response = queryFunction.query(queryRequestCopy);
 
                 if (response.getResults()
                         .size() == 0) {
-                    results = RESULTS_EXHAUSTED;
+                    finished = true;
                     return;
                 }
 
                 results = response.getResults()
                         .iterator();
                 currentIndex += response.getResults()
                         .size();
+
```

Comment:
❓ Since we don't expose the total number of hits anywhere, and since `getHits()` could return `-1` (or could even vary between requests), should we even worry about this condition and simply keep querying until the result list is empty?